src/encoded/static/components/filegallery.js

```diff
@@ -1132,37 +1135,99 @@ VisualizationControls.defaultProps = {
 // Displays the file filtering controls for the file association graph and file tables.
 
 class FilterControls extends React.Component {
-    constructor() {
-        super();
+    constructor(props, reactContext) {
+        super(props);
+
+        this.navigate = reactContext.navigate;
 
         // Bind this to non-React methods.
         this.handleAssemblyAnnotationChange = this.handleAssemblyAnnotationChange.bind(this);
+        this.setDownloadModalVisibility = this.setDownloadModalVisibility.bind(this);
+        this.handleDownloadClick = this.handleDownloadClick.bind(this);
+
+        this.state = { isBatchDownloadOpen: false };
+    }
+
+    setDownloadModalVisibility(isOpen) {
+        this.setState({ isBatchDownloadOpen: isOpen });
     }
 
     handleAssemblyAnnotationChange(e) {
         this.props.handleAssemblyAnnotationChange(e.target.value);
     }
 
+    handleDownloadClick() {
+        const { context, filters } = this.props;
+        const { accession } = context;
+        const type = context && context['@type'] ? context['@type'][0] : 'Experiment';
+        let assemblies = '';
+
+        if (filters.assembly && filters.assembly.length > 0 && filters.assembly[0] !== 'All assemblies') {
+            const analysis = filters.assembly[0].split(' ');
+            const assembly = analysis[0] ? `&files.assembly=${encodedURIComponent(analysis[0])}` : '';
+            const genomeAnnotation = analysis[1] ? `&files.genome_annotation=${encodedURIComponent(analysis[1])}` : '';
+            assemblies = `${[assembly, genomeAnnotation].filter(a => a !== '').join('')}`;
+        }
+        const fileTypes = filters.file_type && filters.file_type.length > 0 ?
+            filters.file_type.map(fileType => `&files.file_type=${encodedURIComponent(fileType)}`).join('') :
+            '';
+        const biologicalReplicates = filters.biological_replicates && filters.biological_replicates.length > 0 ?
```

Comment:
Just FYI because I checked this just a few days ago, AirBnB [seems to recommend](https://github.com/airbnb/javascript#comparison--nested-ternaries) the question mark and colon in ternary expressions at the beginning of the line. ESLint doesn’t check this though.